src/back-end/lib/permissions.ts

```diff
@@ -294,12 +298,12 @@ export async function readSWUProposalScore(connection: Connection, session: Sess
           (proposalStatus === SWUProposalStatus.Awarded || proposalStatus === SWUProposalStatus.NotAwarded) || false));
 }
 
-export function createSWUProposal(session: Session): boolean {
-  return isVendor(session);
+export async function createSWUProposal(connection: Connection, session: Session): Promise<boolean> {
+  return isVendor(session) && await hasAcceptedTerms(connection, session);
 }
 
 export async function submitSWUProposal(connection: Connection, session: Session, organization: Organization): Promise<boolean> {
-  return isVendor(session) && !!session && await isUserOwnerOfOrg(connection, session.user, organization.id) && doesOrganizationMeetSWUQualification(organization);
+  return !!session && isVendor(session) && await hasAcceptedTerms(connection, session) && await isUserOwnerOfOrg(connection, session.user, organization.id) && doesOrganizationMeetSWUQualification(organization);
 }
 
 export async function editSWUProposal(connection: Connection, session: Session, proposalId: string, opportunity: SWUOpportunity): Promise<boolean> {
```

Comment:
I think we'll need to do the check here. If a vendor is editing a submitted proposal before the proposal deadline, they are technically "resubmitting" a proposal. Applies to both CWU and SWU.

Reply:
Added to editing of proposals as well.